### DISCUSSION TITLE
ionospheric correction - how to apply the correction correctly and how to geocode the interferogram?

### DISCUSSION THREAD

**Discussion by Carolinah23** [isce-framework/isce2#563] [Q&A] [answered]
Hello, 

I am using stripmap to create an interferogram with EnviSAT data, so fa I can run everything and create the ionospheric estimation (dispersive). I understand that I should incorporate the correction, but I don't understand how. 

I tried to use the filt_topophase.unw.geo minus the dispersive.bil.unwCor.filt.geo using an external raster calculator. The values of the dispersive file are between -580 and -560, then the corrected interferogram ends with values between 0-530 which I think is not realistic.

Also, I tried to create the corrected interferogram using filt_topophase. unw minus dispersive.bil.filt and then geocoded it again using the geocode function of ISCE. To create the result I use gdal and assign it the ISCE driver. I modify the configuration file to give a list of files to geocode but it says that there are no files to geocode.

I am following the tutorial that is posted by UNAVCO ([2021 InSAR/ISCE Short Course - Day 3](https://www.youtube.com/watch?v=DInVqO0PHpo&ab_channel=UNAVCO%2CInc.)), but Heresh just rewrapped and plotted the results but did not create the unwrapped and geocoded final file. 

Thanks for any help!

**Comment by fdelgadodelapuente** [comment, marked as answer]
Hi Carolinah23,

there is no need to run the split spectrum correction with ENVISAT data because C-band is not significantly affected by ionospheric perturbations, except for the ESD correction in zones of low geomagnetic latitudes as described here,
https://ieeexplore.ieee.org/document/8706258

TO apply the split spectrum correction for ALOS data you can adapt the following example for the stripmap stack processor

        imageMath.py -e='a_0;a_1*(c>0)-b_0*(c>0)' -s BIL --a=interferogram/filt_topophase.unw --b=ionosphere/dispersive.bil.unwCor.filt  -o interferogram/filt_topophase_nondispersive.unw --c=interferogram/filt_topophase.conncomp 
 
you then add the the corrected interferogram to the list of files in geocode list

        <property name="geocode list">["interferogram/filt_topophase_nondispersive.unw"]</property>

**Reply by Carolinah23** [reply]
Oh great, thanks for your answer!
I will try in the future the code for ALOS.

I was wondering how to interpret the results tho. I ran the workshop example in the video and I got values of <-550 in the dispersive file, why are these so large?
For the tropospheric corrections I also got huge values but per image, then when I do the difference between the tropospheric effects of the reference and the secondary image I got small values and I can remove them from the interferogram and the original results don't change that much.

**Reply by fdelgadodelapuente** [reply]
It's just the reference level. These corrections can be off by several tens of radians but the key thing is that they remove the long wavelength signals. Afterwards you just reference your corrected interferogram to a stable non-deforming area.

**Reply by Carolinah23** [reply]
Thanks, I think I understand! 
Just to clarify the last part; if my stable non-deforming area has a value of -500 rad after correction (I am doing just tropospheric because I am using EnviSAT, but let's say tropospheric and ionospheric) I must subtract -500 rad from all the pixels in the interferogram to have the real displacement with reference to that area, right?

**Reply by fdelgadodelapuente** [reply]
Yes. You can do it in two ways. 

First, pick a small area that has little to none phase delays, calculate the mean value and subtract it from the interferogram.  

Second, if your inferred deforming area only covers a fraction of the interferogram, like for volcanoes, you can remove a linear ramp and that will reference the interferogram to the average of the far field signal. If the whole area is deforming like for interseismic strain studies, then you cannot remove a ramp blindly.

**Reply by Carolinah23** [reply]
I think I will have to do it. My study area should have small deformations (a few mm/y) although each interferogram, that cover between 1-3 months, shows large displacements, and these increase when I apply the tropospheric correction. 
Thank you again for your help!

**Reply by fdelgadodelapuente** [reply]
Those "large displacements" are most likely strong atmospheric phase delays. If they increase after applying the tropospheric correction, then the correction is not predicting the data and you should not use it. The rate of success of tropospheric correction calculated from global atmospheric models varies from region to region and should be tested case by case.

**Reply by Carolinah23** [reply]
I see, it makes sense. In one of the interferograms, I got a mean displacement of -0.707 cm which is enormous because there are only 35 days between the two images. After the tropospheric correction, the mean displacement is 14.2 cm.
I am using GACOS and reading the papers they posted on their website, but I will appreciate any resource you find helpful!
Thanks for your explanation!
